pvlib/tests/test_solarposition.py

```diff
@@ -673,6 +673,28 @@ def test_hour_angle():
     assert np.allclose(hours, expected)
 
 
+def test_hour_angle_with_tricky_timezones():
+    # tests timezones that have a DST shift at midnight
```

Comment:
I'd like to add two more checks that should fail, to detect if `pandas.normalize()` changes its behavior. I think this works for the first case:

```
with pytest.raises(pytz.exceptions.NonExistentTimeError):
    times.normalize()
```

Reply:
Good idea.  I just added 2 tests like that in a new commit.